src/leo_s3_user.erl

```
@@ -161,35 +194,43 @@ import(UserId, AccessKey, SecretKey) ->
+
+    %% insert a user-credential-info
+    case mnesia:write(?AUTH_TABLE,
+                      #credential{access_key_id = AccessKey,
+                                  secret_access_key = SecretKey,
+                                  created_at = CreatedAt}, write) of
+        ok ->
+            %% insert a user-info
+            case mnesia:write(?USERS_TABLE,
+                              #?S3_USER{id = UserId,
+                                        password = <<>>,
+                                        created_at = CreatedAt,
+                                        updated_at = CreatedAt}, write) of
                 ok ->
-                    case leo_s3_user_credential:put(#user_credential{user_id = UserId,
-                                                                     access_key_id = AccessKey,
-                                                                     created_at = CreatedAt}) of
+                    %% insert a user-credential-info
+                    case mnesia:write(?USER_CREDENTIAL_TABLE,
+                                      #user_credential{user_id = UserId,
+                                                       access_key_id = AccessKey,
+                                                       created_at = CreatedAt}, write) of
                         ok ->
-
-                            {ok, Keys};
-                        Error ->
-                            Error
+                            {ok, [{access_key_id,     AccessKey},
+                                  {secret_access_key, SecretKey}]};
+                        _ ->
+                            {error, transaction_abort}
                     end;
-                Error ->
-                    Error
+                _ ->
+                    {error, transaction_abort}
             end;
-        Error ->
-            Error
+        _ ->
+            {error, transaction_abort}
```

As I said at the above comment, I'd recommend you write the code in the `let it crash` way to make the code simple and readable while the rollback is properly called in case any errors happen.